### ISSUE TITLE
ffmpeg/avconv was not found (and I couldn't install it)

### ISSUE THREAD

**Issue by alessio-ds** [Just-Some-Bots/MusicBot#1279]
`Traceback (most recent call last):
  File "/usr/local/lib/python3.5/asyncio/tasks.py", line 239, in _step
    result = coro.send(None)
  File "/root/musicbotto/musicbot/player.py", line 255, in _play
    after=lambda: self.loop.call_soon_threadsafe(self._playback_finished)
  File "/usr/local/lib/python3.5/site-packages/discord/voice_client.py", line 434, in create_ffmpeg_player
    raise ClientException('ffmpeg/avconv was not found in your PATH environment variable') from e
discord.errors.ClientException: ffmpeg/avconv was not found in your PATH environment variable`

I obviously followed the "Guide for Ubuntu" on github, I also tried to install ffmpeg in other ways, like following this guide https://trac.ffmpeg.org/wiki/CompilationGuide/Ubuntu, but it still gives me errors.
Any suggestions? 
I'm using Ubuntu 14.04.

**Comment by chuushi** [comment]
Why not just run `sudo apt-get install ffmpeg`?

**Comment by alessio-ds** [comment]
Well...this is quite obvious but


`root@alessiosca:~/musicbotto# apt-get install ffmpeg`
`Reading package lists... Done`
`Building dependency tree`
`Reading state information... Done`
`Package ffmpeg is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source`
`E: Package 'ffmpeg' has no installation candidate`

**Comment by dude22072** [comment]
14.04 removed ffmpeg support, you wont be able to get it through a normal sudo apt-get. It was added back in 15 and onwards, though. Try this if you need to stay on 14: https://www.faqforge.com/linux/how-to-install-ffmpeg-on-ubuntu-14-04/

**Comment by alessio-ds** [comment]
I can't even do the first step!

`root@alessiosca:~# sudo add-apt-repository ppa:mc3man/trusty-media`
`sudo: unable to execute /usr/bin/add-apt-repository: No such file or directory`

**Comment by dude22072** [comment]
whoops, forgot 14 doesn't come with PPA support. Do "sudo apt-get install software-properties-common" then try that link again.

**Comment by alessio-ds** [comment]
still the same...
sudo: unable to execute /usr/bin/add-apt-repository: No such file or directory

btw, doing your command it runs out an error:

`root@alessiosca:~# sudo apt-get install software-properties-common`
`Reading package lists... Done`
`Building dependency tree`
`Reading state information... Done`
`The following packages were automatically installed and are no longer required:
  libavdevice53 libavfilter3 libavresample1 libcdio-cdda1 libcdio-paranoia1
  libcdio13 libsdl1.2debian`
`Use 'apt-get autoremove' to remove them.`
`The following extra packages will be installed:`
 ` python3-software-properties`
`The following packages will be upgraded:`
`  python3-software-properties software-properties-common`
`2 upgraded, 0 newly installed, 0 to remove and 112 not upgraded.`
`581 not fully installed or removed.`
`Need to get 0 B/40.0 kB of archives.`
`After this operation, 1,024 B of additional disk space will be used.`
`Do you want to continue? [Y/n] y`
`(Reading database ... 97047 files and directories currently installed.)`
`Preparing to unpack .../software-properties-common_0.92.37.8_all.deb ...`
`/var/lib/dpkg/info/software-properties-common.prerm: 6: /var/lib/dpkg/info/software-properties-common.prerm: py3clean: not found`
`dpkg: warning: subprocess old pre-removal script returned error exit status 127`
`dpkg: trying script from the new package instead ...`
`/var/lib/dpkg/tmp.ci/prerm: 6: /var/lib/dpkg/tmp.ci/prerm: py3clean: not found`
`dpkg: error processing archive /var/cache/apt/archives/software-properties-common_0.92.37.8_all.deb (--unpack):`
` subprocess new pre-removal script returned error exit status 127`
`/var/lib/dpkg/info/software-properties-common.postinst: 6: /var/lib/dpkg/info/software-properties-common.postinst: py3compile: not found`
`dpkg: error while cleaning up:`
` subprocess installed post-installation script returned error exit status 127`
`Preparing to unpack .../python3-software-properties_0.92.37.8_all.deb ...`
`/var/lib/dpkg/info/python3-software-properties.prerm: 6: /var/lib/dpkg/info/python3-software-properties.prerm: py3clean: not found`
`dpkg: warning: subprocess old pre-removal script returned error exit status 127`
`dpkg: trying script from the new package instead ...`
`/var/lib/dpkg/tmp.ci/prerm: 6: /var/lib/dpkg/tmp.ci/prerm: py3clean: not found`
`dpkg: error processing archive /var/cache/apt/archives/python3-software-properties_0.92.37.8_all.deb (--unpack):`
 `subprocess new pre-removal script returned error exit status 127`
`/var/lib/dpkg/info/python3-software-properties.postinst: 6: /var/lib/dpkg/info/python3-software-properties.postinst: py3compile: not found`
`dpkg: error while cleaning up:`
` subprocess installed post-installation script returned error exit status 127`
`Errors were encountered while processing:`
 `/var/cache/apt/archives/software-properties-common_0.92.37.8_all.deb`
 `/var/cache/apt/archives/python3-software-properties_0.92.37.8_all.deb`
`E: Sub-process /usr/bin/dpkg returned an error code (1)`




tl;dr 
`Errors were encountered while processing:`
 `/var/cache/apt/archives/software-properties-common_0.92.37.8_all.deb`
 `/var/cache/apt/archives/python3-software-properties_0.92.37.8_all.deb`
`E: Sub-process /usr/bin/dpkg returned an error code (1)`

can't understand how to fix this

**Comment by jayktaylor** [comment]
Best solution to this would be to update to Ubuntu 16.
